### PR TITLE
[WIP] ENH: Add --dismiss-t1w option to skip anatomical processing

### LINKED ISSUE #755: Running only functionals
A good option would be to run FMRIPREP in the absence of T1w images under two possible circumstances:
  - The dataset does not have T1w images
  - The user set a flag `--dismiss-t1w` to force dismissing the anatomical processing.

(comes from [this discussion](https://neurostars.org/t/running-only-functionals/855?u=oesteban))

### PR DESCRIPTION
**Pull request by effigies**
Closes #755.

Replaces #765.

### CODE REVIEW

**Comment by effigies** [comment]
Hi @madeleinesnyder. Looks like you got your commit up, so I went ahead and opened a PR from your branch. If you go to [Files changed](https://github.com/poldracklab/fmriprep/pull/770/files), you'll be able to see the differences from our tree. This is just a sort of orientation/house-keeping post, before getting into the details of your PR.

Any changes you make to your branch will show up in this PR, so if you want to work on it, make sure you're in this branch:

```Shell
git checkout dismiss_t1`
```

And then when you `git push`, they'll show up here.

Let's make sure that you have an `upstream` remote. If you run `git remote -v | grep upstream`, and you see

```Shell
upstream	https://github.com/poldracklab/fmriprep.git (fetch)
upstream	https://github.com/poldracklab/fmriprep.git (push)
```

Then you're good. If you see something different, run:

```
git remote set-url upstream https://github.com/poldracklab/fmriprep.git
```

And if you don't see anything at all, run:

```
git remote add upstream https://github.com/poldracklab/fmriprep.git
```

If you want to work on other things, I'd recommend starting other branches. If it's another feature you want to submit, you can check out from the latest `master`:

```Shell
git fetch upstream
git checkout -b new_feature_branch upstream/master
git branch -u origin
```

Where `new_feature_branch` is some descriptive name. When you `git push`, it will push to a branch of the same name on your repository (`origin`).

Let me know if you have any questions, here.

**Comment by effigies** [comment]
Now I'm going to walk you through this status box:

conflicts

The first thing you'll see will be the big red Xs. These are really handy for us to quickly see what needs attention, but contributors can find them discouraging, so I'd encourage you not to worry about them. Sometimes it will be a problem on our end. We'll tell you if we need you to do something.

The next thing is the "This branch has conflicts that must be resolved". What this means is that there are changes we've made since you forked that overlap with some changes we've made. Since there have been a lot on our end, I'm going to go ahead and resolve the conflicts on my machine, and I'll send you a  pull request against your branch. You'll be able to review that and discuss what I've done. And then, if you merge it, it will update this pull request, and it will show that there aren't conflicts.

File: fmriprep/workflows/base.py [hunk 1]
```diff
@@ -65,7 +66,6 @@ def init_fmriprep_wf(subject_list, task_id, run_uuid,
                               output_spaces=['T1w', 'fsnative',
                                             'template', 'fsaverage5'],
                               template='MNI152NLin2009cAsym',
-                              medial_surface_nan=False,
```
Comment (oesteban) [inline fmriprep/workflows/base.py]: This line should not be removed, sorry I left it in my PR.

File: fmriprep/workflows/base.py [hunk 2]
```diff
@@ -236,11 +237,11 @@ def init_single_subject_wf(subject_id, task_id, name,
                                     template='MNI152NLin2009cAsym',
                                     output_spaces=['T1w', 'fsnative',
                                                   'template', 'fsaverage5'],
-                                    medial_surface_nan=False,
```
Comment (oesteban) [inline fmriprep/workflows/base.py]: Same again, this line should be here.

File: fmriprep/workflows/base.py [hunk 3]
```diff
@@ -295,8 +298,6 @@ def init_single_subject_wf(subject_id, task_id, name,
              - fsaverage (or other pre-existing FreeSurfer templates)
         template : str
             Name of template targeted by `'template'` output space
-        medial_surface_nan : bool
```
Comment (oesteban) [inline fmriprep/workflows/base.py]: These two lines, again.

File: fmriprep/workflows/base.py [hunk 4]
```diff
@@ -374,77 +376,99 @@ def init_single_subject_wf(subject_id, task_id, name,
...
-                                               freesurfer=freesurfer,
-                                               use_bbr=use_bbr,
-                                               bold2t1w_dof=bold2t1w_dof,
-                                               reportlets_dir=reportlets_dir,
+    if not func_only:
+        anat_preproc_wf = init_anat_preproc_wf(name="anat_preproc_wf",
+                                               skull_strip_template=skull_strip_template,
                                                output_spaces=output_spaces,
                                                template=template,
-                                               medial_surface_nan=medial_surface_nan,
```
Comment (oesteban) [inline fmriprep/workflows/base.py]: This line (only this one) again was removed mistakenly.

**Comment by oesteban** [comment]
Hi @madeleinesnyder, those commits don't look very good. Seems like you had some changes commited in your local branch that were not pushed to your remote (and I couldn't see those).

Do you mind if I overwrite this PR history?

**Comment by madeleinesnyder** [comment]
I think I just fixed it-- I remerged my branch

**Comment by oesteban** [comment]
Have you pushed the changes? If so, I think it didn't work...

**Comment by madeleinesnyder** [comment]
I haven't changed anything personally, just resolved the conflicts on the
dismiss_t1 branch on my fork of your repo (which was the PR)

**Comment by madeleinesnyder** [comment]
Sorry for all this github trouble-- I think it might be better to just
reclone the repo and start over from what you have so far if that would be
less of a headache

**Comment by oesteban** [comment]
Hehehe, no problem, let me rewrite the history and I'll tell you how to ensure you are at the tip of the repo.

**Comment by madeleinesnyder** [comment]
Ok thanks so much!

**Comment by oesteban** [comment]
Ok, I don't have permissions to modify your branch. Let's open a brand-new PR:

You'll need to take the following steps:

1. Check you have the appropriate remotes. Please check that the output of the command is close enough to what I show you here.

```Bash
$ git remote -v
origin	git@github.com:madeleinesnyder/fmriprep.git (fetch)
origin	git@github.com:madeleinesnyder/fmriprep.git (push)
upstream	https://github.com/poldracklab/fmriprep.git (fetch)
upstream	https://github.com/poldracklab/fmriprep.git (push)
```
If you don't have the two entries that start with upstream, run:
```
git remote add upstream https://github.com/poldracklab/fmriprep.git
```
If you have an `upstream` here, but the URL is different

```
git remote set-url upstream https://github.com/poldracklab/fmriprep.git
```
Once you have a correct `upstream`, then add my repo and update it:

```
git remote add oesteban https://github.com/oesteban/fmriprep.git
git fetch oesteban
```

2. Let's get your local copy updated
```
git fetch upstream
git checkout master
git merge upstream/master
git push
```
Now, you can go to your [GitHub repo](https://github.com/madeleinesnyder/fmriprep) and check that your master is even with poldracklab:master, an example from other projects: https://msdnshared.blob.core.windows.net/media/MSDNBlogsFS/prod.evol.blogs.msdn.com/CommunityServer.Blogs.Components.WeblogFiles/00/00/00/60/29/metablogapi/4201.image5_39F0BA75.png

3. Restart the branch:

```
git checkout -b func-only oesteban/enh/func-only
git push --set-upstream origin func-only
```

4. Submit the PR through GitHub: I think [this url](https://github.com/poldracklab/fmriprep/compare/master...madeleinesnyder:func-only?expand=1) should work if everything went alright.

**Comment by madeleinesnyder** [comment]
Alright I followed those instructions (thank you for being so thorough) and
submitted the PR.

**Comment by oesteban** [comment]
Yay!

**Comment by madeleinesnyder** [comment]
Is it correct that there are no func-only flags or parameters in this new
updated branch?

**Comment by oesteban** [comment]
I can see [this addition](https://github.com/madeleinesnyder/fmriprep/blob/86c010eff01bfe535e608e47cf3ae4fdd7a3c64e/fmriprep/cli/run.py#L92-L94). Possibly clearer in the [diff view](https://github.com/poldracklab/fmriprep/pull/808/files). Is that what you meant?

**Comment by oesteban** [comment]
BTW, let's keep conversations within the new PR - https://github.com/poldracklab/fmriprep/pull/808

**Comment by madeleinesnyder** [comment]
Ah yes, got it. I see now there is a whole new workflow to take care of the
func-only case instead of editing the original func workflow.

Also, I was thinking that it might be helpful for me to contribute a
BIDSgenerator file that would be easily configured by people who want to
use the pipeline but don't have the BIDS data format. I know our lab was
hesitant to use the pipeline because of the overhead of renameing and
restructuring data, so it might increase use if there is an add-on file
that takes care of this. I could put in the places in run.py and other
scripts where BIDSgenerator.py is called

**Pull request closed without merging**